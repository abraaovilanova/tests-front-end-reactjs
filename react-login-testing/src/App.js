import React from 'react'
import Login from './pages/Login'
import Dashboard from './pages/Dashboard'
import PropTypes from 'prop-types'

import './App.css';

function App({ isLoggedin, username, products }) {
  return (
    <div className="App">
        {
          isLoggedin
          ? (
            <Dashboard username={username} products={products} />
          )
          : (
            <Login />
          )
        }
    </div>
  );
}

// App.propTypes = {
//   isLoggedin: PropTypes.bool,
//   username: PropTypes.string,
//   products: PropTypes.array,
// }

export default App;
