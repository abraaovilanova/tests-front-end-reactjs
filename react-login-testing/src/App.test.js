import React from "react"
import App from "./App"
import { mount } from "enzyme"
import toJson from "enzyme-to-json"

it("should render the Login for the non logged in user", () => {
  const props = {
    isLoggedin: false,
    username: null,
    products: [],
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()
})

it("should render the Dashboard for the logged in user", () => {
  const props = {
    isLoggedin: true,
    username: "Pedro",
    products: [
      {
        id: 1,
        name: "T-shirt",
        price: 25,
      },
      {
        id: 2,
        name: "Pants",
        price: 50,
      },
    ],
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()
})

it("should render the Dashboard for the logged in user without Products", () => {
  const props = {
    isLoggedin: true,
    username: "Joana",
    products: null,
  }
  const app = mount(<App {...props} />)

  expect(toJson(app)).toMatchSnapshot()
})

// import React from "react"
// import App from "./App"
// import { mount } from "enzyme"
// import Login from "./pages/Login"
// import Dashboard from "./pages/Dashboard"

// it("should render the Login for the non logged in user", () => {
//   const props = {
//     isLoggedin: false,
//     username: null,
//     products: [],
//   }
//   const app = mount(<App {...props} />)

//   expect(app.find(Dashboard)).toHaveLength(0)
//   expect(app.find(Login)).toHaveLength(1)
//   expect(app.find("strong").at(0).text()).toEqual(
//     "Hello stranger. Sign in to continue."
//   )
//   expect(app.find("button").at(0).text()).toEqual("Sign")
// })

// it("should render the Dashboard for the logged in user", () => {
//   const props = {
//     isLoggedin: true,
//     username: "Pedro",
//     products: [
//       {
//         id: 1,
//         name: "T-shirt",
//         price: 25,
//       },
//       {
//         id: 2,
//         name: "Pants",
//         price: 50,
//       },
//     ],
//   }
//   const app = mount(<App {...props} />)

//   expect(app.find(Dashboard)).toHaveLength(1)
//   expect(app.find(Login)).toHaveLength(0)

//   expect(app.find("p").at(0).text()).toEqual("Welcome, Pedro!")

//   expect(app.find("li")).toHaveLength(2)
//   expect(app.find("li").at(0).text()).toEqual("Name: T-shirt|Price: 25")
//   expect(app.find("li").at(1).text()).toEqual("Name: Pants|Price: 50")
// })

// it("should render the Dashboard for the logged in user without Products", () => {
//   const props = {
//     isLoggedin: true,
//     username: "Joana",
//     products: null,
//   }
//   const app = mount(<App {...props} />)

//   expect(app.find(Dashboard)).toHaveLength(1)
//   expect(app.find("p").at(0).text()).toEqual("Welcome, Joana!")
//   expect(app.find("strong").at(0).text()).toEqual(
//     "Joana has no registered products"
//   )
// })