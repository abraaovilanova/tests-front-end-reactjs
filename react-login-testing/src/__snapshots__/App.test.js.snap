// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Dashboard for the logged in user 1`] = `
<App
  isLoggedin={true}
  products={
    Array [
      Object {
        "id": 1,
        "name": "T-shirt",
        "price": 25,
      },
      Object {
        "id": 2,
        "name": "Pants",
        "price": 50,
      },
    ]
  }
  username="Pedro"
>
  <div
    className="App"
  >
    <_default
      products={
        Array [
          Object {
            "id": 1,
            "name": "T-shirt",
            "price": 25,
          },
          Object {
            "id": 2,
            "name": "Pants",
            "price": 50,
          },
        ]
      }
      username="Pedro"
    >
      <div>
        <p>
          Welcome, 
          Pedro
          !
        </p>
        <ul>
          <li
            key="1"
          >
            <p>
              Name: 
              T-shirt
            </p>
            |
            <p>
              Price: 
              25
            </p>
          </li>
          <li
            key="2"
          >
            <p>
              Name: 
              Pants
            </p>
            |
            <p>
              Price: 
              50
            </p>
          </li>
        </ul>
        <strong>
          Total:
          75
        </strong>
      </div>
    </_default>
  </div>
</App>
`;

exports[`should render the Dashboard for the logged in user without Products 1`] = `
<App
  isLoggedin={true}
  products={null}
  username="Joana"
>
  <div
    className="App"
  >
    <_default
      products={null}
      username="Joana"
    >
      <div>
        <p>
          Welcome, 
          Joana
          !
        </p>
        <strong>
          Joana
           has no registered products
        </strong>
      </div>
    </_default>
  </div>
</App>
`;

exports[`should render the Login for the non logged in user 1`] = `
<App
  isLoggedin={false}
  products={Array []}
  username={null}
>
  <div
    className="App"
  >
    <_default>
      <div>
        <strong>
          Hello stranger. Sign in to continue.
        </strong>
        Username:
        <input />
        Passoword: 
        <input />
        <button
          onClick={[Function]}
        >
          Sign
        </button>
      </div>
    </_default>
  </div>
</App>
`;
